internal/trace/transport.go

```diff
@@ -87,3 +93,52 @@ func logHeader(header http.Header) string {
 	}
 	return "   Empty header"
 }
+
+// logResponseBody prints out the response body if it is printable and within
+// the size limit.
+func logResponseBody(resp *http.Response) string {
+	if resp.Body == nil || resp.Body == http.NoBody {
+		return "   No response body to print"
+	}
+
+	// non-applicable body is not printed and remains untouched for subsequent processing
+	contentType := resp.Header.Get("Content-Type")
+	if !isPrintableContentType(contentType) {
+		return fmt.Sprintf("   Response body of content type \"%s\" is not printed", contentType)
+	}
+
+	// read the body up to limit+1 to check if the body exceeds the limit
+	lr := io.LimitReader(resp.Body, payloadSizeLimit+1)
+	readBody, err := io.ReadAll(lr)
+	if err != nil {
+		return fmt.Sprintf("   Error reading response body: %v", err)
+	}
+	// restore the body by concatenating the read body with the remaining body
+	resp.Body = io.NopCloser(io.MultiReader(bytes.NewReader(readBody), resp.Body))
```

Comment:
The original `resp.Body` is never closed even if later `resp.Body.Close()` is called.

Reply:
Good catch! Fixed.